bokehjs/src/lib/models/glyphs/varea_step.ts

```diff
@@ -90,43 +90,71 @@ export class VAreaStepView extends AreaView {
     return [scx, scy]
   }
 
-  protected override _hit_point(geometry: PointGeometry): Selection {
+  protected _line_selection_for(i: number): Selection {
+    const result = new Selection()
+    result.add_to_selected_glyphs(this.model)
+    result.view = this
+    result.line_indices = [i]
+    return result
+  }
+
+  protected _hit_point_before(geometry: PointGeometry): Selection {
     const {sx, sy1, sy2} = this
+    for (let i = 1; i < this.data_size; i++) {
+      const px = [sx[i-1], sx[i], sx[i], sx[i-1]]
+      const py = [sy1[i], sy1[i], sy2[i], sy2[i]]
+      if (hittest.point_in_poly(geometry.sx, geometry.sy, px, py)) {
+        return this._line_selection_for(i)
+      }
+    }
+    return new Selection()
+  }
 
+  protected _hit_point_after(geometry: PointGeometry): Selection {
+    const {sx, sy1, sy2} = this
     for (let i = 0; i < this.data_size - 1; i++) {
-      let px: number[]
-      let py: number[]
+      const px = [sx[i], sx[i+1], sx[i+1], sx[i]]
+      const py = [sy1[i], sy1[i], sy2[i], sy2[i]]
+      if (hittest.point_in_poly(geometry.sx, geometry.sy, px, py)) {
+        return this._line_selection_for(i)
+      }
+    }
+    return new Selection()
+  }
 
-      switch (this.model.step_mode) {
-        case "before":
-          px = [sx[i], sx[i+1], sx[i+1], sx[i]]
-          py = [sy1[i+1], sy1[i+1], sy2[i+1], sy2[i+1]]
-          break
-        case "after":
-          px = [sx[i], sx[i+1], sx[i+1], sx[i]]
-          py = [sy1[i], sy1[i], sy2[i], sy2[i]]
-          break
-        case "center":
-        {
-          const mid_x = (sx[i] + sx[i+1]) / 2
-          px = [sx[i], mid_x, mid_x, sx[i+1], sx[i+1], mid_x, mid_x, sx[i]]
-          py = [sy1[i], sy1[i], sy1[i+1], sy1[i+1], sy2[i+1], sy2[i+1], sy2[i], sy2[i]]
-          break
-        }
+  protected _hit_point_center(geometry: PointGeometry): Selection {
+    const {sx, sy1, sy2} = this
+    for (let i = 0; i < this.data_size; i++) {
+      let px: number[]
+      const mid_prev_x = (sx[i-1] + sx[i]) / 2
```

Comment:
@solstag this seems to be computing `sx[i-1]` even when `i=0` ?

Reply:
It is, but that's the intended semantics: `mid_prev_x` for `i == 0` is undefined, and not used in that case. The same is true for `mid_next_x` on line 130 when `i == this.data_size -1`.

Reply:
I do understand that it's harmless in the sense that subsequent code manually coordinates to not-use the undefined values at just the right times. My concern was that it seems potentially fragile from a maintenance POV. Someone in the future might rearrange or refactor the code and then the undefined cases end up accidentally being used when they should not. But two lines of duplication would also be a slight maintenance wart as well, so I suppose it is a toss-up.